src/Microsoft.AspNet.Authentication.OAuthBearer/OAuthBearerAuthenticationOptions.cs

```diff
@@ -54,15 +46,14 @@ public OAuthBearerAuthenticationOptions() : base()
         /// <summary>
         /// Gets or sets the challenge to put in the "WWW-Authenticate" header.
         /// </summary>
-        /// TODO - brentschmaltz, should not be null.
-        public string Challenge { get; set; }
+        public string Challenge { get; [param: NotNull] set; } = OAuthBearerAuthenticationDefaults.AuthenticationScheme;
 
         /// <summary>
         /// The object provided by the application to process events raised by the bearer authentication middleware.
         /// The application may implement the interface fully, or it may create an instance of OAuthBearerAuthenticationProvider
         /// and assign delegates only to the events it wants to process.
         /// </summary>
-        public OAuthBearerAuthenticationNotifications Notifications { get; set; }
+        public OAuthBearerAuthenticationNotifications Notifications { get; [param: NotNull] set; } = new OAuthBearerAuthenticationNotifications();
```

Comment:
Okay. `param: NotNull` is currently used in a few places. Do you want me to remove them?

https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationOptions.cs#L41
https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication/Notifications/RedirectToIdentityProviderNotification.cs#L26